lib/TPP/Transforms/IntelAMXTileConfigHoisting.cpp

```diff
@@ -0,0 +1,101 @@
+//===- IntelAMXTileConfigHoisting.cpp ----------------------------*- C++-*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements tile configuration hoisting on parallel loops.
+//
+//===----------------------------------------------------------------------===//
+#include "TPP/Dialect/Xsmm/XsmmOps.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+namespace mlir {
+namespace tpp {
+#define GEN_PASS_DEF_INTELAMXTILECONFIGHOISTINGPASS
+#include "TPP/Passes.h.inc"
+} // namespace tpp
+} // namespace mlir
+
+using namespace mlir;
+using namespace mlir::xsmm;
+
+namespace mlir {
+namespace tpp {
+
+struct IntelAMXTileConfigHoisting : OpRewritePattern<memref::AllocaOp> {
```

Comment:
nit: seems a bit brittle to anchor it on alloca, I think it'd be safer to target `IntelAMXTileConfigOp`

Reply:
I agree with both of you. This is not critical and not high prio. I have opened #896. Let's discuss a good solution over there and we track the open there.